src/blant.c

```diff
@@ -1033,14 +1044,15 @@ int main(int argc, char *argv[])
 		if (_windowRep_min_num_edge < 0) Fatal("WindowRep minimum number of edges must be larger than 0. Check edge density\n");
     }
 
-    if (_windowRep_unambig || _sampleMethod == SAMPLE_INDEX){
-        _windowRep_unambig_set = SetAlloc(_numCanon);
-        SET *orbit_temp = SetAlloc(_numOrbits);
-        for(i=0; i<_numCanon; i++) if SetIn(_connectedCanonicals, i)
-        {
+    _windowRep_allowed_ambig_set = SetAlloc(_numCanon);
+    SET *orbit_temp = SetAlloc(_numOrbits);
+    for(i=0; i<_numCanon; i++) {
+        if (SetIn(_connectedCanonicals, i)) {
             // calculate number of permutations for the given canonical graphlet (loop through all unique orbits and count how many times they appear)
             // the formula is for every unique orbit, multiply the number of permutations by the factorial of how many appearances that unique orbit has
             // if there is one orbit with three nodes and a second orbit with 2 nodes, the number of permutations would be (3!)(2!)
+            // NOTE: this is not the most efficient algorithm since it doesn't use hash tables. I didn't want to overcomplicate it because it only happens once per run.
```

Comment:
Probably don't need hash tables; that's the point of a bit-wise set implementation. (And if the creation happens only once per run it's totally fine.)